src/Microsoft.Diagnostics.ExtensionCommands/FindPointersInCommand.cs

```diff
@@ -0,0 +1,497 @@
+                        values.AddRange(item.Value);
+                    else
+                        destDict[item.Key] = new(item.Value);
+                }
+            }
+        }
+
+        private class MemoryWalkContext
+        {
+            private readonly Dictionary<ulong, (string, int)> _resolved = new();
+            private readonly ClrObject[] _pinned;
+
+            public MemoryWalkContext(IEnumerable<ClrObject> pinnedObjects)
+            {
+                _pinned = pinnedObjects.Where(o => o.IsValid && !o.IsFree).OrderBy(o => o.Address).ToArray();
+            }
+
+            public bool IsPinnedObject(ulong address, out ClrObject found)
+            {
+                ClrObject last = _pinned.LastOrDefault();
+                if (_pinned.Length == 0 || address < _pinned[0].Address || address >= last.Address + last.Size)
+                {
+                    found = default;
+                    return false;
+                }
+
+                int low = 0;
+                int high = _pinned.Length - 1;
+                while (low <= high)
+                {
+                    int mid = (low + high) >> 1;
+                    if (_pinned[mid].Address + _pinned[mid].Size <= address)
+                    {
+                        low = mid + 1;
+                    }
+                    else if (address < _pinned[mid].Address)
+                    {
+                        high = mid - 1;
+                    }
+                    else
+                    {
+                        found = _pinned[mid];
+                        return true;
+                    }
+                }
+
+                found = default;
+                return false;
+            }
+
+            public (string Symbol, int Offset) ResolveSymbol(IModuleService modules, ulong ptr)
+            {
+                if (_resolved.TryGetValue(ptr, out (string, int) result))
+                    return result;
+
+                // _resolved is just a cache.  Don't let it get so big we eat all of the memory.
+                if (_resolved.Count > 16 * 1024)
+                    _resolved.Clear();
+
+                IModule module = modules.GetModuleFromAddress(ptr);
+                IModuleSymbols symbols = module?.Services.GetService<IModuleSymbols>();
+
+                if (symbols is not null && symbols.TryGetSymbolName(ptr, out string symbolName, out ulong displacement))
+                {
+                    string moduleName = module.FileName;
+                    if (!string.IsNullOrWhiteSpace(moduleName))
+                        symbolName = Path.GetFileName(moduleName) + "!" + symbolName;
+
+                    return _resolved[ptr] = (symbolName, displacement > int.MaxValue ? int.MaxValue : (int)displacement);
+                }
+
+                return (null, -1);
+            }
+        }
+
+        [HelpInvoke]
+        public string GetDetailedHelp()
+        {
+            return
```

Comment:
The GetDetailedHelp functions return void and have to do a WriteLine(...) around the string. Ironically what you have is where I want to end up (and allow the function to be static) but these help changes are in a future PR.